File: python/process.py
"""process.py: Code used by more than one sorting file.
Contains functionality to:
	Generate a list of random numbers in a range
	Swap two items in an array in place (mostly)
	Initialize global variables
"""

__author__ = "Lucas Burns"
__version__ = "2017-9-26"

import random

def populate(len = 10, min = 1, max = 100):
	items = []
	for i in range(0, len):
		items.append(random.randint(min, max))
	return items

def swap(first, second, items):
	if first in range(0,len(items)) and second in range(0,len(items)):
                temp = items[first]
		items[first] = items[second]
		items[second] = temp

def initCounters():
	# Number of times we have compared two values for sorting
	global comparisons
	comparisons = 0
	# Number of times we have read a value in or out of a comparison
	global accesses
	accesses = 0
	# Number of times we have written to an array
	global writes
	writes = 0

File: ruby/process.rb
# Contains code used by more than one ruby sorting file.
# Functionality includes: 
# 	Generation of a list of random numbers
# 	Swapping of two items in an array in place (mostly)

module Process
	def populate(len = 10, min = 1, max = 100)
		items = []
		for i in 0..len do
			items << rand(min..max)
		end
		return items
	end

	def swap(left, right, items)
		if left.between?(0, items.length - 1) && right.between?(0, items.length - 1)
			temp = items[left]
			items[left] = items[right]
			items[right] = temp
		end
	end
end